S175 initial version

davidfq opened the pull request:
See task for details (comments on modules structure): [Tenant API: Terraform AWS modules](https://app.asana.com/0/501803088048006/1207456044599358/f).

davidfq committed:
Add GitHub basic workflows

davidfq committed:
Add modules: auth and tenant API script

davidfq committed:
Add basic example

davidfq committed:
Add ignore rules

davidfq committed:
Fix workflow

davidfq committed:
Lint code

davidfq committed:
Add READMEs for each submodule/example

davidfq committed:
Update API path

davidfq commented on README.md:
```diff
@@ -1,2 +1,19 @@
+See the [examples/create_psoxy_connections] example for a complete usage example.
+
+[Psoxy Connections]: https://docs.worklytics.com/psoxy
+[Tenant API]: https://docs.worklytics.com/knowledge-base/tenant-api
```
This doesn't exit yet; but I will create a placeholder soon

aperez-worklytics commented on .github/workflows/terraform_validate.yml:
```diff
@@ -0,0 +1,30 @@
+        with:
+          terraform_version: ${{ matrix.terraform_version }}
+
+      - name: 'Terraform - validate examples/create_psoxy_connections'
```
No validation for `modules` ? When the example is link to a published version of the module, I think we need something to validate the actual code we have there.

aperez-worklytics commented on modules/cognito_tenant_api_auth/main.tf:
```diff
@@ -0,0 +1,38 @@
+}
+
+resource "aws_iam_policy" "cognito_developer_identities" {
+  name        = "${aws_cognito_identity_pool.worklytics_identity_pool.identity_pool_name}_CognitoDeveloperIdentity"
```
Something like 
```suggestion
  name        = "${aws_cognito_identity_pool.worklytics_identity_pool.identity_pool_name}_CognitoDeveloperIdentityForWorklyticsTerraform"
```
?

aperez-worklytics commented on modules/psoxy_connection/main.tf:
```diff
@@ -0,0 +1,29 @@
+resource "local_file" "worklytics_tenant_api_script_file" {
```
This is going to create the file to be executed, but then we need to run it ourselves right?

Something it happens when I was doing [workload identity in psoxy](https://github.com/Worklytics/psoxy/pull/242/commits/495cfa9b7bd1d0e5ce7b9cb9060ba18c99d6816b#r1088197220).

I had to put the file as external and invoke it through `data` . With that, the script is executed as part of apply -and users don't need to do more interactions with TF-

davidfq replied:
Yeah, the problem is that they have to set up the `DataConnectionAdmin` user principal on our side before the script execution. I guess that we can document the process and tell them to configure that upfront

davidfq committed:
Update modules/cognito_tenant_api_auth/main.tf

Co-authored-by: aperez-worklytics <75276364+aperez-worklytics@users.noreply.github.com>

davidfq committed:
Merge remote-tracking branch 'origin/S175-initial-version' into S175-initial-version

davidfq committed:
Validate modules

davidfq committed:
Fix script module

- Add extra settings (bulk and github)
- Make script filename variable (multiple connections to the same source)
- Delete `kind` setting (AWS context given)

davidfq committed:
Fix example

- Missing tenant API host
- Define extra settings

davidfq committed:
Fix format

davidfq committed:
Remove Terraform older versions

Optional attributes (`module_variable_optional_attrs`) was
an experimental feature in 1.1.x - 1.2.x versions, and our
official Psoxy docs explicitly states that Terraform 1.3.x
is a requirement

davidfq committed:
Fix links

davidfq committed:
Add comment about variable

davidfq commented:
@jlorper this is ready; fully tested (see https://github.com/Worklytics/evalengin/pull/6934)

davidfq committed:
Reference example from submodules

davidfq committed:
Fix grammar

davidfq committed:
Highlight requisite

jlorper commented on .github/workflows/terraform_lint.yaml:
```diff
@@ -0,0 +1,28 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        terraform_version: [ '~1.5.0', '~1.6.0', '~1.7.0', 'latest'] # in theory, we go down to 1.3, but I think that's overkill for lint
```
starting rc-0.5 this is >= 1.6
https://github.com/Worklytics/psoxy/blob/rc-v0.5.0/.github/workflows/ci-terraform-modules.yaml
So adapt everywhere please

jlorper commented on README.md:
```diff
@@ -1,2 +1,19 @@
+The main use-case for these submodules is to create [Psoxy Connections] in Worklytics via its [Tenant API]. 
+
+After a successful Psoxy deployment (Data Source fully configured, Psoxy instance tested and running), the last step
+should be to create the actual connection in the [Data Sources] section of the Worklytics Web App. These submodules
+provide a way to automate this process.
```
```suggestion
should be to create the actual connection in the [Data Sources] section of the Worklytics Web App, either manually following the instructions in the TO-DO files, or automatically with a script that is produced by these sub-modules.
```
not totally sold on this... what do you think?

jlorper commented on examples/create_psoxy_connections/variables.tf:
```diff
@@ -0,0 +1,44 @@
+  description = "The email of the user that has been granted access to the Worklytics Tenant API (configure in Worklytics Web App)."
+}
+
+variable "psoxy_connections" {
```
but this should be provided by regular proxy output, if the customer needs to fill this list by hand is a nightmare and not useful at all.
I thought this reused the output of the [worklytics connector](https://github.com/Worklytics/psoxy/blob/rc-v0.5.0/infra/modules/worklytics-psoxy-connection-generic/main.tf) but if that is not the case, it is not very useful

jlorper reviewed (commented):
I don't get how the connectors are filled in, if it is manual (which appears to be) is a waste of time for the customer. We should reuse the proxy output.

davidfq committed:
Define example outputs

davidfq committed:
Add missing setting: bucket

davidfq committed:
Improve script generation

davidfq committed:
Update example
